### DISCUSSION TITLE
How can I temporarily disable an option set in gitconfig for the time of running a command?

### DISCUSSION THREAD

**Discussion by nimser** [dandavison/delta#1209] [Q&A] [answered]
The documentation says an option can be enabled at run time using the `DELTA_FEATURES` environment variable (e.g. `git diff HEAD HEAD^1 | DELTA_FEATURES='+side-by-side' delta`).
However, say I already have the `side-by-side` activated in my gitconfig and I want to disable for the time of running that command. How would I do that? I've tried changing `+` to `-` like in `git diff HEAD HEAD^1 | DELTA_FEATURES='-side-by-side' delta` and other things to no avail.

**Comment by dandavison** [comment, marked as answer]
Hi @nimser, one way is

```
git -c delta.side-by-side=false <gitsubcommand>
```

Another way might be to enable it in your gitconfig, not directly using `delta.side-by-side`, but rather in a feature, and then use `DELTA_FEATURES` to disable that feature on-demand (let me know if that's not clear). That might not be convenient though if you're using other features.

Do you think there's a role for `-` to mean subtract a feature? It's not implemented currently but if you think it makes sense do feel free to experiment and open a PR.

**Reply by nimser** [reply]
Thanks, it works! About the "Another way" perhaps you meant "use `DELTA_FEATURES` to ~disable~ **enable** that feature on-demand"? Because if I understand there's no way to *disable* a feature with `DELTA_FEATURES`?

**Reply by dandavison** [reply]
Well, if you don't use `+` then you can use the env var to overwrite the feature list, so you could remove a feature, and if that feature was the way that `side-by-side` was being activated in your config then it would deactivate `side-by-side`, right? But, that's not going to be very convenient if we're using multiple features. So I was wondering whether your suggestion of adding `-` functionality to the env var made sense.

**Reply by nimser** [reply]
Makes sense now. The `-` feature would have a use case though your `git -c` approach satisfies my needs (for now). If I find myself fiddling too much with the env var I'll consider the PR, thanks!
